Develop

turkycat opened the pull request:
Adding SetworkSerial class and also fixed a bug in UsbSerial where the program will crash when the device service is not found or accessible.

zfields commented on source/Serial/NetworkSerial.cpp:
```diff
@@ -0,0 +1,228 @@
+		}
+		catch( Platform::Exception ^e )
+		{
+			ConnectionFailed( ref new Platform::String( L"BluetoothSerial::connectAsync failed with a Platform::Exception type. Message: " ) + e->Message );
```
BluetoothSerial:: s/b NetworkSerial::

zfields commented on source/Serial/NetworkSerial.cpp:
```diff
@@ -0,0 +1,228 @@
+
+NetworkSerial::NetworkSerial(
+	Windows::Networking::HostName ^host,
+	uint16_t port
```
`host_` and `port_` for the sake of consistency

zfields commented on source/Serial/NetworkSerial.h:
```diff
@@ -0,0 +1,113 @@
+	[Windows::Foundation::Metadata::DefaultOverload]
+	NetworkSerial(
+		Windows::Networking::HostName ^host,
+		uint16_t port
```
host_  and  port_  for the sake of consistency

zfields commented on source/Serial/NetworkSerial.h:
```diff
@@ -0,0 +1,113 @@
+	Concurrency::task<void>
+		connectToHostAsync(
+			Windows::Networking::HostName ^host,
+			uint16_t port
```
host_  and  port_  for the sake of consistency

zfields commented on source/Serial/NetworkSerial.cpp:
```diff
@@ -0,0 +1,228 @@
+Concurrency::task<void>
+NetworkSerial::connectToHostAsync(
+	Windows::Networking::HostName ^host,
+	uint16_t port
```
host_  and  port_  for the sake of consistency

zfields commented on source/Serial/NetworkSerial.cpp:
```diff
@@ -0,0 +1,228 @@
+	void
+	)
+{
+	end();
```
fire off a `ConnectionLost();` here

zfields commented on source/Serial/NetworkSerial.cpp:
```diff
@@ -0,0 +1,228 @@
+	}
+}
+
+/// \details Immediately discards the incoming parameters, because they are used for standard serial connections and will have no bearing on a bluetooth connection.
```
Make this comment relevant to networking

turkycat commented:
Good comments. I have corrected the clarity and consistency issues.
I also cleaned up a few more potential crashes for null objects and changes the destructor's behavior with ConnectionLost.
